src/Database/RocksDB/Base.hs

```diff
@@ -127,10 +135,34 @@ createSnapshotBracket db = allocate (createSnapshot db) (releaseSnapshot db)
 --
 -- The returned handle should be released with 'close'.
 open :: MonadIO m => FilePath -> Options -> m DB
-open path opts = liftIO $ bracketOnError (mkOpts opts) freeOpts mkDB
+open path opts = liftIO $ bracketOnError initialize finalize mkDB
     where
-        mkDB opts'@(Options' opts_ptr _ _) =
-            withCString path $ \path_ptr ->
+# ifdef mingw32_HOST_OS
+        initialize =
+            (, ()) <$> mkOpts opts
+        finalize (opts', ()) =
+            freeOpts opts'
+# else
+        initialize = do
+            opts' <- mkOpts opts
+            -- With LC_ALL=C, two things happen:
+            --   * rocksdb can't open a database with unicode in path;
+            --   * rocksdb can't create a folder properly.
```

Comment:
1. Is the first problem solved in this PR?
2. If it is, isn't that solution enough for the second problem?